engine/src/main/java/org/pentaho/di/job/Job.java

```diff
@@ -460,6 +458,8 @@ private void emergencyWriteJobTracker( Result res ) {
    */
   private Result execute() throws KettleException {
     try {
+      this.logBufferStartLine = KettleLogStore.getAppender().getLastBufferLineNr();
```

Comment:
Why not just call setInitialLogBufferStartLine()? In future, you could need to add extra code to that method

Reply:
Changed as suggested